File: ruby/src/Creational/Factory_Method/BostonSoupFactoryMethodSubclass.rb


#//BostonSoupFactoryMethodSubclass - Two of Two Subclass Factory Methods
require "Creational/Factory_Method/SoupFactoryMethod"
require "common/BostonClamChowder"
require "common/BostonFishChowder"

class BostonSoupFactoryMethodSubclass < SoupFactoryMethod

	def makeBuffetName()
   		return "Boston Soup Buffet"
	end

	def makeClamChowder()
   		return BostonClamChowder.new()
	end

	def makeFishChowder()
   		return BostonFishChowder.new()
	end

end

File: csharp/src/Creational/Factory_Method/BostonSoupFactoryMethodSubclass.cs
//BostonSoupFactoryMethodSubclass.java - One of Two Subclass Factory Methods

namespace Creational.Factory_Method {

using common;

public class BostonSoupFactoryMethodSubclass : SoupFactoryMethod {
    override public string makeBuffetName() {
        return "Boston Soup Buffet";
    }
    override public ClamChowder makeClamChowder() {
        return new BostonClamChowder();
    }
    override public FishChowder makeFishChowder() {
        return new BostonFishChowder();
    }
}

}
